### ISSUE TITLE
Any valid resource can be Posted in any resource URL endpoint

### ISSUE THREAD

**Issue by nirojshrestha019** [awslabs/fhir-works-on-aws-deployment#682]
**Describe the bug**
I Posted Organization resources to POST `<baseURL>/Patient` endpoint and got 201 responses. 

**To Reproduce**
1. Go to <baseURL>/Patient` endpoint and add the following body of Organization resource:
```
{
  "resourceType": "Organization",
  "name": "Health Level Seven International",
  "alias": [
    "HL7 International"
  ],
  "telecom": [
    {
      "system": "phone",
      "value": "(+1) 734-677-7777"
    },
    {
      "system": "fax",
      "value": "(+1) 734-677-6622"
    },
    {
      "system": "email",
      "value": "hq@HL7.org"
    }
  ],
  "address": [
    {
      "line": [
        "3300 Washtenaw Avenue, Suite 227"
      ],
      "city": "Ann Arbor",
      "state": "MI",
      "postalCode": "48104",
      "country": "USA"
    }
  ]
}
```
2. We get 201 responses. Here is the screenshot:
![image](https://user-images.githubusercontent.com/28623129/186726006-4e8cd24d-62c9-4d20-beb9-b85eb9b922fb.png)

**Expected behavior**
I expect it to not allow Organization resources to be posed in /Patient endpoint (i.e., only allow resource to be posted in that URL endpoint.)

**Screenshots**

**Versions (please complete the following information):**
 - v3.0.0-smart

**Comment by maghirardelli** [comment]
Thank you for bringing this to our attention! We are looking into this and will get back to you!

**Comment by ssvegaraju** [comment]
Hi @nirojshrestha019, this is actually expected behavior, weird as it may look. We forcibly set the `resourceType` field when sending requests to match the route specified in the POST request. So the `Observation` resourceType in the JSON above would've been rewritten to match the `Patient` endpoint. This means that the resource is validated as a Patient resource as expected. However, the properties in the JSON object above actually work for both Observation and Patient resources, so the request succeeds and a Patient resource is created. I would suggest adding fields unique to the Observation resource to ensure that validation fails if this behavior isn't desired (i.e. the `subject` property). Hope this helps!

Thanks for your time,
Sukeerth

**Comment by nirojshrestha019** [comment]
Thanks, @ssvegaraju. The issue is not only with the Observation resource allowed in /Patient endpoint but also other resources like organization as well. This expected behavior is kinda weird and I still think it's a bug. 
Ideally, all the endpoints should only be allowed to POST expected resources, isn't it (i.e., /Patient endpoint should only allow Patient resource to post and likewise for other endpoints as well.)? This avoids a lot of confusion and adds one more checkpoint for the data validation.

**Comment by ssvegaraju** [comment]
That's a fair point, I'll take this back to the team and put it on our backlog!

**Comment by ElliotSilver** [comment]
If you make the change @nirojshrestha019 suggests, then there is no longer a need to forcibly set the resourceType field, since the resource would be rejected if the resourceType didn't match.

**Comment by ssvegaraju** [comment]
Hi @nirojshrestha019, @ElliotSilver. Just updating that a fix for this has been merged, and will be available with the next release of the routing package! (We are currently on v6.6.0, this should be available on v6.6.1+). I'll go ahead and close the issue for now, but feel free to reopen if there's anything else!

Edit: v6.6.1 is up on npm here: https://www.npmjs.com/package/fhir-works-on-aws-routing
